packages/core/strapi/lib/services/entity-service/index.js

```diff
@@ -213,8 +214,10 @@ const createDefaultImplementation = ({ strapi, db, eventHub, entityValidator })
       return null;
     }
 
+    const componentsToDelete = await getComponents(uid, entityToDelete);
```

Comment:
I'm wondering if we could just populate the entityToDelete above when doing the findOne so we avoid doing the getComponents manually.

Reply:
That would be optimal! I will try to populate them in the findOne

Reply:
@alexandrebodin 
To recap. This could be dangerous:
```js
const entityToDelete = await db.query(uid).findOne({
      ...query,
      where: { id: entityId },
      populate: getComponentAttributes(strapi.getModel(uid)),
    });
```
entityToDelete is returned to the user, so this can bring two problems:

- If `populate` was already used (in query), we might introduce conflicts by overriding it.
- Returning the entity with populated attributes which were not asked for.

For the moment, I am more inclined to leave the current implementation as valid. 
One alternative I can think of is doing a [_.merge()](https://www.geeksforgeeks.org/lodash-_-merge-method/). I don't know if that could solve it for us.

Reply:
Yes thanks for sharing what we discussed in DM 👍 

However I think it is worth it to implement the TODO at  packages/generators/app/lib/create-project.js L791 
`// TODO: support multiple relations at once with the populate syntax` so you can just call the load function once and let it handle all the loading. 

If that's too much we can keep it for later. wdyt ?

Reply:
that's sounds great to me.

I will try to look what does this imply, but I think It would be too much now.
I will add the TODO 🙂

Reply:
@alexandrebodin 
Are you sure that is the correct file? 
Shouldn't it be in the query builder / entity manager?

Reply:
Nevermind, I guess you were refering to `packages/core/database/lib/entity-manager.js`

Reply:
Yes mb

Reply:
@alexandrebodin 

Ok! Have some questions.
- When exactly are you supposed to use the `populate` and `load` functions? The only place I see `load` being used is to load components. Is that it?
- What are you proposing for this PR is, using the `load()` or `populate()` function instead of `findOne()`. Where you can pass an array of populates, that should be merged into a single `populate` parameter?
Something like this?
```js
const populate = mergeDeep(
  { "dynamiczone": { populate: {"a": true}}}, 
  {
    "relation": true,
    "dynamiczone": { populate: {"b": true}},
    "multiple_media": { populate: ['folder']},
  }
 );
```

Reply:
> @alexandrebodin
> 
> Ok! Have some questions.
> 
> * When exactly are you supposed to use the `populate` and `load` functions? The only place I see `load` being used is to load components. Is that it?

- I don't think we use `populate` I kept it as a way to add some relation to an already loaded entry. 

- `load`. Is used also for example in graphql to load relations :) 

In your case you could use both depending on the rest of the implementation :)


> * What are you proposing for this PR is, using the `load()` or `populate()` function instead of `findOne()`. Where you can pass an array of populates, that should be merged into a single `populate` parameter?
>   Something like this?
> 
> ```js
> const populate = mergeDeep(
>   { "dynamiczone": { populate: {"a": true}}}, 
>   {
>     "relation": true,
>     "dynamiczone": { populate: {"b": true}},
>     "multiple_media": { populate: ['folder']},
>   }
>  );
> ```

So I'm proposing to keep the findOne as is to do something like:

```js
const entity = findOne(...)

const componentData = load(uid, entity, ['compA', 'compB', ...])

deleteEntity(entity)
deleteComponents(componentData)
```

Reply:
🆒 Got it. I will try to do that! Thank you

Reply:
@alexandrebodin Ok! I think I got something working.

- I updated the entity-manager `load` to now receive an array or a single element.
   - To avoid breaking changes, when passing a single element you receive an object with component attributes:
      { "id": value , "compoAttribute": value}
   - When passing an array , you receive an object like:
      { "compo_name":  { "id": value , "compoAttribute": value}}